youtube_dl/extractor/sbs.py

```diff
@@ -46,9 +78,191 @@ class SBSIE(InfoExtractor):
+        if errnote is False:
+            return False
+        if errnote is None:
+            errnote = 'Unable to download webpage'
+
+        errmsg = '%s: %s' % (errnote, error_to_compat_str(err))
+        if fatal:
+            raise ExtractorError(errmsg, sys.exc_info()[2], cause=err, video_id=video_id)
+        else:
+            self._downloader.report_warning(errmsg)
+            return False
+
+    def _download_webpage_handle(self, url, video_id, *args, **kwargs):
+        # note, errnote, fatal, encoding, data, headers={}, query, expected_status
+        # specialised to detect geo-block
+
+        errnote = args[2] if len(args) > 2 else kwargs.get('errnote')
+        fatal = args[3] if len(args) > 3 else kwargs.get('fatal')
+        exp = args[7] if len(args) > 7 else kwargs.get('expected_status')
+
+        exp = variadic(exp or [], allowed_types=(compat_str, ))
+        if 403 not in exp and '403' not in exp:
+            exp = list(exp)
+            exp.append(403)
+        else:
+            exp = None
+
+        if exp:
+            if len(args) > 7:
+                args = list(args)
+                args[7] = exp
+            else:
+                kwargs['expected_status'] = exp
+                kwargs = compat_kwargs(kwargs)
+
+        ret = super(SBSIE, self)._download_webpage_handle(url, video_id, *args, **kwargs)
+        if ret is False:
+            return ret
+        webpage, urlh = ret
+
+        if urlh.getcode() == 403:
+            if urlh.headers.get('x-error-reason') == 'geo-blocked':
+                countries = ['AU']
+                if fatal:
+                    self.raise_geo_restricted(countries=countries)
+                err = GeoRestrictedError(
+                    'This Australian content is not available from your location due to geo restriction',
+                    countries=countries)
+            else:
+                err = compat_HTTPError(urlh.geturl(), 403, 'HTTP Error 403: Forbidden', urlh.headers, urlh)
+            ret = self.__handle_request_webpage_error(err, video_id, errnote, fatal)
+
+        return ret
+
+    def _extract_m3u8_formats(self, m3u8_url, video_id, *args, **kwargs):
+        # ext, entry_protocol, preference, m3u8_id, note, errnote, fatal,
+        # live, data, headers, query
+        entry_protocol = args[1] if len(args) > 1 else kwargs.get('entry_protocol')
+        if not entry_protocol:
+            entry_protocol = 'm3u8_native'
+            if len(args) > 1:
+                args = list(args)
+                args[1] = entry_protocol
+            else:
+                kwargs['entry_protocol'] = entry_protocol
+                kwargs = compat_kwargs(kwargs)
+
+        return super(SBSIE, self)._extract_m3u8_formats(m3u8_url, video_id, *args, **kwargs)
+
+    _AUS_TV_PARENTAL_GUIDELINES = {
+        'P': 0,
+        'C': 7,
+        'G': 0,
+        'PG': 0,
+        'M': 14,
```

Comment:
This was my source: https://rating-system.fandom.com/wiki/Australian_TV_content_rating_system

I had some trouble working out what it meant for M and MA15+. Definitely add NC.

Let's remove the `_` at the start of the class variable name and then it can be imported if required by one of the handful of other .au sites.

Reply:
> I had some trouble working out what it meant for M and MA15+. Are all these 15+?

M means it's recommended for people 15+
MA15+ means that it's strictly 15+ (cinemas ask for ID, for example)

> It doesn't mention X18+ either: == R18+ ?

X18+ is similar to R18+, but it's reserved for sexually explicit content (see https://www.classification.gov.au/classification-ratings/what-do-ratings-mean#X18+)


On a side note, it seems you were actually right, there is an MAV15+ category (see https://www.alrc.gov.au/publication/classification-content-regulation-and-convergent-media-alrc-report-118/9-classification-categories-and-criteria-2/classification-categories/)